patches/v2.0/0001-llvm-cov-add-Clang-s-Source-based-Code-Coverage-supp.patch

```diff
@@ -0,0 +1,647 @@
+From 3b71080443fee47743b1a43476a5815772b638a8 Mon Sep 17 00:00:00 2001
+From: Wentao Zhang <wentaoz5@illinois.edu>
+Date: Sun, 25 Aug 2024 17:56:02 -0500
+Subject: [PATCH v2 1/5] llvm-cov: add Clang's Source-based Code Coverage
+ support
+
+Implement debugfs entries for serializing profiles and resetting
+counters/bitmaps. Also adds Source-based Code Coverage flags and kconfig
+options.
+
+This work reuses a portion of code from a previous effort by Sami
+Tolvanen et al. [1], specifically its debugfs interface and the
+underlying profile processing, but discards all its PGO-specific parts,
+notably value profiling.  To our end (code coverage, like MC/DC,
+required for high assurance), we do instrumentation at the compiler
+frontend, instead of IR; we care about counters and bitmaps, but not
+value profiling.
+
+[1] https://lore.kernel.org/lkml/20210407211704.367039-1-morbo@google.com/
+
+Signed-off-by: Chuck Wolber <chuck.wolber@boeing.com>
+Signed-off-by: Wentao Zhang <wentaoz5@illinois.edu>
+---
+ Makefile                          |   3 +
+ arch/Kconfig                      |   1 +
+ include/asm-generic/vmlinux.lds.h |  36 +++++
+ kernel/Makefile                   |   1 +
+ kernel/llvm-cov/Kconfig           |  47 ++++++
+ kernel/llvm-cov/Makefile          |   5 +
+ kernel/llvm-cov/fs.c              | 253 ++++++++++++++++++++++++++++++
+ kernel/llvm-cov/llvm-cov.h        | 156 ++++++++++++++++++
+ scripts/Makefile.lib              |  11 ++
+ scripts/mod/modpost.c             |   2 +
+ 10 files changed, 515 insertions(+)
+ create mode 100644 kernel/llvm-cov/Kconfig
+ create mode 100644 kernel/llvm-cov/Makefile
+ create mode 100644 kernel/llvm-cov/fs.c
+ create mode 100644 kernel/llvm-cov/llvm-cov.h
+
+diff --git a/Makefile b/Makefile
+index 7b60eb103..19faf5a31 100644
```

Comment:
Sorry I forgot to specify `--abbrev=12` to `git format-patch`. Will update this together with other possible changes in the next pass.

Reply:
Do you see any evidence that it matters? The only feedback I have seen if hash length is too small. Otherwise the `format-patch` defaults in the most recent versions of Git do not appear to be creating any problems for submitters.

Reply:
I don't actually know. I sampled a few recent mailing list patches and they seem to use 12. I guess longer SHA-1 won't hurt at least.

Reply:
In that case, I would stick to defaults unless we learn otherwise.